report_thesis/src/sections/proposed_approach/testing_validation.tex

```diff
@@ -47,76 +61,142 @@ \subsection{Validation and Testing Procedures for Model Evaluation}\label{subsec
 This necessitates careful dataset partitioning to ensure that the model training process accounts for these challenges, improving the generalizability and robustness of the models.
 
 \subsubsection{Dataset Partitioning}\label{subsubsec:dataset_partitioning}
-To ensure rigorous evaluation of our models and to address the challenges of data leakage and uneven distribution of extreme values, we have implemented a customized k-fold cross-validation procedure.
-This procedure ensures that all data points from a given target are either entirely in the training set or the test set, overcoming the challenge of data leakage we mention above.
-Additionally, it ensures that extreme values are handled appropriately to maintain an even distribution across folds, as demonstrated by the data distributions in Figure~\ref{fig:oxide_distributions}.
+To ensure rigorous evaluation of our models and to address the challenges of data leakage and uneven distribution of extreme values, we have implemented a customized k-fold data partitioning procedure. 
+This approach divides the dataset into $k$ folds, which it uses to define cross-validation data sets, as well as a training set and a test set.
+The procedure ensures that all data points from a given target are only present in one of the $k$ folds, overcoming the challenge of data leakage we mention above.
+Additionally, it ensures that extreme values are handled by redistributing them evenly across the training folds, preventing any single fold from being disproportionately influenced by these values.
 
 \begin{algorithm}
 \caption{Data Partitioning With Extreme Value Handling}
 \label{alg:custom_kfold_cv}
 \begin{algorithmic}[1]
-\Require Dataset \( \mathbf{D} \), group column \( g \), target column \( t \), number of splits \( k \), percentile \( p \), random seed \( \textit{seed} \)
-\Ensure Cross-validation folds \( \mathbf{F}_\text{cv} \), training set \( \mathbf{D}_\text{train} \), and test set \( \mathbf{D}_\text{test} \)
-\State \label{line:seed} Set random seed for reproducibility if \(\text{seed} \) is not None
-\State \label{line:remove_duplicates} Remove duplicate entries based on \( g \) and sort by \( t \)
-\State \label{line:assign_folds} Assign fold numbers sequentially from 0 to \( k-1 \) to unique targets
+\Require Dataset $\mathbf{D}$, group column $g$, target column $t$, number of splits $k$, percentile $p$, random seed $\textit{seed}$
+\Ensure Training and validation sets for cross-validation $\mathbf{T}_\text{cv}$, training set $\mathbf{D}_\text{train}$, and test set $\mathbf{D}_\text{test}$
```

Comment:
Modified to clarify that we don't return 'folds,' which is ambiguous as to what we mean - is it indices? data? ranges?

So now we explicitly clarify that it's data partitions